lib/query/builder.js

```diff
@@ -565,6 +566,21 @@ assign(Builder.prototype, {
     return this._aggregate('avg', column);
   },
 
+  // Retrieve the "count" of the distinct results of the query.
+  countDistinct: function countDistinct() {
+    return this._aggregateDistinct(true).count.apply(this, arguments);
```

Comment:
Instead call `return this._aggregate('sum', column, true)`, and have `_aggregate` take a distinct flag.